src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -166,7 +166,7 @@ void RealmCoordinator::set_config(const Realm::Config& config)
         }
 
         if (config.sync_config) {
-            if (m_config.sync_config->user != config.sync_config->user) {
+            if (*m_config.sync_config->user != *config.sync_config->user) {
```

Comment:
should we check that the user is not null here? It being null on either side is an error. Not sure if we check for that elsewhere.

Reply:
@jbreams No tests blew up without.

We do not allow people to open flexible sync configuration without a user in Dart, and I assume in any other SDKs, even if core somehow allow it.

That said, this is not a hot path, and we can be as defensive as we like. If you prefer I will add the check, however it is unclear to me if it is okay for both to be a shared nullptr? As the code stood previously that was okay, but to me it seems an error. What is your take?

Reply:
My take is that we're dereferencing some pointers without checking that they aren't null and we could segfault if this method is called incorrectly. If the answer is that we check elsewhere that these aren't null and are always safe to de-reference then that's fine, but we should check _somewhere_ the they aren't null before we try to access them.

Reply:
Sorry, lost the ball here - other stuff came up. 

I have pushed a commit that checks  for `null` before the comparison.

Should we error out, if one (or both) of the `shared_ptr`s are `null`? Not sure it can happen, but if it happens would that be an error? As is the code it just avoids the potential crash, but doesn't raise an exception.

Something is rotten on Jenkins, but it is not related to this PR, as best I can tell.